documentation/projects/proposals/relicensing/20241028-implementation_plan_relicensing.md

```diff
@@ -0,0 +1,301 @@
+
+### Determine new licenses
+
+- Libraries
+
+  The GNU LGPL license allows our libs to be used universally in both free as
+  well as non-free environments.
+
+- Applications
+
+  The GNU AGPL license requires derivatives of the API and frontend, which are
+  accessed over a network to also share source code, which is want we want.
+
+- Everything else
+
+  The GNU GPL license is a good fit for the rest of the codebase, since there
+  are no special requirements for this code (unlike apps and libs).
+
+### Institute code freeze
+
+We need to institute a code freeze because we need to add the commit hash of the
+last commit that will be under the MIT license to the `LICENSING.md` file.
+
+Any PRs opened before the code freeze can take one of three routes.
+
+- Be merged before the license transition and be covered by the MIT license
+- Reaffirm that they are compatible with the GNU license family and be merged.
+- Be closed without being merged.
+
+### Apply new licenses
+
+To apply different licenses to different parts of the repository, we will take
+the following steps.
+
+#### Add a `LICENSING.md` file
+
+The purpose of this file is to explain what licenses apply to which code block
+and how they are cascaded. It should look like this template (3 fields need to
+be filled out):
+
+```md
+The default license for this project is
+[GPL-3.0-or-later](https://spdx.org/licenses/GPL-3.0-or-later.html).
+
+The following libraries are licensed under
+[LGPL-3.0-or-later](https://spdx.org/licenses/LGPL-3.0-or-later.html) and not
+under the default license.
+
+- <libraries' names go here>
+
+The following applications are licensed under
+[AGPL-3.0-or-later](https://spdx.org/licenses/AGPL-3.0-or-later.html) and not
+under the default license.
+
+- <applications' names go here>
+
+## History
+
+All code before <commit> is available under
+[MIT](https://spdx.org/licenses/MIT.html).
+```
+
+#### Add/update `LICENSE` files
+
+We will update the existing root license and add the appropriate licenses for
+the code blocks that were exempted from the default and applied specific
+licenses.
+
+- `api/LICENSE` (new, AGPL-3.0-or-later)
+- `frontend/LICENSE` (new, AGPL-3.0-or-later)
+- `packages/js/api-client/LICENSE` (new, LGPL-3.0-or-later)
+- `packages/python/openverse-attribution/LICENSE` (new, LGPL-3.0-or-later)
+- `LICENSE` (MIT → GPL-3.0-or-later)
+
+#### Update license in manifests
+
+The following files contain references to the project license. These will need
+to be updated.
+
+- Python packages
+  - `.vale/pyproject.toml`
+  - `api/pyproject.toml`
+  - `documentation/pyproject.toml`
+  - `indexer_worker/pyproject.toml`
+  - `packages/python/openverse-attribution/pyproject.toml`
+- JS packages
+  - `packages/js/api-client/package.json`
+  - `packages/js/k6/package.json`
+- Other references
+  - `documentation/packages/js/api_client/index.md`
+  - `api/conf/settings/spectacular.py`
+  - `catalog/tests/factories/github/pull.json`
+
+## Communications
```

Comment:
Will the Make blog be the medium for these announcements?

Reply:
I think a Make blog announcement is necessary, where we should clarify that this change does not affect the consumers of the API and users of the frontend in any way. Once the license change is done, we can also push it via marketing channels and social media, as it is good to have more consolidation and alignment within the WP ecosystem.